src/Views/HardwareView.vala

```diff
@@ -280,6 +282,73 @@ public class About.HardwareView : Gtk.Grid {
 
         return 0;
     }
+
+    private string get_storage_type () {
+        string partition_name = get_partition_name ();
+        string disk_name = get_disk_name (partition_name);
+        string path = "/sys/block/%s/queue/rotational".printf(disk_name);
+        string storage = "";
+        var rotational_storage_file = File.new_for_path (path);
+        try {
+            var dis = new DataInputStream (rotational_storage_file.read ());
+            string line;
+            while ((line = dis.read_line ()) != null) {
+                if (line.has_prefix ("0")) {
+                    storage = "(SSD)";
+                } else {
+                    storage = "(HDD)";
+                }
+            }
+        } catch (Error e) {
+            warning (e.message);
+        }
+        return storage;
+    }
+
+    private string get_partition_name () {
+        string df_stdout;
+        string df_stderr;
+        string partition = "";
+        int df_status;
+        try {
+            Process.spawn_command_line_sync ("df /",
+                out df_stdout,
+                out df_stderr,
+                out df_status);
+            string[] output = df_stdout.split("\n");
+            foreach(string line in output) {
+                if (line.has_prefix ("/dev/")) {
+                    string[] partition_line = line.split(" ");
```

Comment:
Space before `(`.

I guess if the line starts with `/dev/` you can safely assume that you can strip everything else after the first space and not bother with a foreach loop.

You could do e.g this:
```
int idx = line.index_of (" ");
if (idx != -1) {
    partition = line.substring (0, idx);
}
```